backend/Dockerfile

```diff
@@ -13,32 +26,40 @@ RUN apt update && apt install -y \
     gcc \
     && rm -rf /var/lib/apt/lists/*
 
-# install python libraries (except torch)
-COPY requirements/ requirements/
 ENV PIP_CERT=$CACERT_LOCATION
-RUN pip --default-timeout=300 install --upgrade pip \
-    && pip --default-timeout=300 install --no-cache-dir -r requirements/common.txt \
-    && rm -r /root/.cache
+
+# Download dependencies as a separate step to take advantage of Docker's caching.
+# Leverage a cache mount to /root/.cache/pip to speed up subsequent builds.
+# Leverage a bind mount to requirements.txt to avoid having to copy them into
+# into this layer.
+RUN --mount=type=cache,target=/root/.cache/pip \
+    --mount=type=bind,source=requirements.txt,target=requirements.txt \
+    python -m pip install -r requirements.txt
 
 ARG VERSION
 ARG MODEL="EffB7_2023-03-06_08"
 ENV SSL_CERT_FILE=$CACERT_LOCATION
 RUN curl -o model.pth https://storage.gra.cloud.ovh.net/v1/AUTH_df731a99a3264215b973b3dee70a57af/basegun-public/models/${MODEL}/${MODEL}.pth
-COPY src/ src/
+COPY . .
 RUN mkdir -p src/weights \
     && mv model.pth src/weights/model.pth \
     && echo '{"app": "'${VERSION}'", "model": "'${MODEL}'"}' > versions.json
 
-# launch website
+# Copy the source code into the container.
+COPY . .
+
+# Expose the port that the application listens on.
+EXPOSE 8000
+
+# Install pytorch
 FROM base as dev
-RUN pip --default-timeout=300 install --no-cache-dir -r requirements/dev.txt
+RUN pip install torch==1.13.0 torchvision==0.14.0 --extra-index-url https://download.pytorch.org/whl/cpu

```

Comment:
I'm really not a big fan to write requirements outside of a requirements folder or txt, it's not a good practice to my mind.
Any python developper when looking at a repo will have the reflex to check the requirements to see the dependancies, and won't especially think of checking inside the Dockerfile.